### PR TITLE
blob url in source of qna pairs 

### PR DESCRIPTION
**Pull request by akhilayesantarao**
- add blob path excluding common path as source in qna maker
- use path to find document corresponding to source

### CODE REVIEW

File: CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs [hunk 1]
```diff
@@ -54,7 +55,7 @@ public static class UploadToQnAMaker
             {
                 return new BadRequestObjectResult($"{skillName} - Invalid request record array.");
             }
-            var msgBatch = new QnAQueueMessageBatch(); 
+            var msgBatch = new QnAQueueMessageBatch();
```
Comment (bingisbestest) [inline CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs:58]: Remove additional space, also we should have style cop to catch such issues.

File: CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs [hunk 2]
```diff
@@ -220,19 +231,18 @@ private static async Task<string> CreateKB(QnAMakerClient qnaClient, ILogger log
...
-        private static async Task<Operation> UpdateKB(QnAMakerClient qnaClient, UpdateKbOperationDTO updateKB, string kbId, ILogger log)
+        private static async Task<Operation> UpdateKB(QnAMakerClient qnaClient, string updateKB, string kbId, ILogger log)
         {
             var service = "/qnamaker/v4.0";
             string uri = qnaClient.Endpoint + service + "/knowledgebases/" + kbId;
             var dummyOperation = new Operation(operationState: OperationStateType.Failed);
 
-            string content = JsonConvert.SerializeObject(updateKB);
             // To speed up extraction within one batch request and to skip sources that have extraction failure
             string append = ",\"maxDegreeOfParallelism\":4, \"skipSourcesWithExtractionFailure\":true";
```
Comment (bingisbestest) [inline CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs:241]: can we have one method to do all this string concatenation

File: CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs [hunk 3]
```diff
@@ -83,7 +84,8 @@ public static class UploadToQnAMaker
                         {
                             Id = id,
                             FileName = blobName,
-                            FileUri = fileUri
+                            FileUri = fileUri,
+                            BlobPath = blobUrl.Remove(0, blobBaseURL.Length)
```
Comment (bingisbestest) [inline CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs:88]: why can't we do this on the file Uri won't that be a blob url with a SAS?

Comment (akhilayesantarao) [reply CustomSkillForDataIngestion/QnAIntegrationCustomSkill/QnAIntegrationCustomSkill.cs:88]: sas token would expire right

**Review by dereklegenzoff — APPROVED** [review]

**Review by bingisbestest — APPROVED** [review]

**Pull request merged**